app/ante/eth.go

```diff
@@ -213,8 +215,11 @@ func (egcd EthGasConsumeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simula
 		gasPool.ConsumeGas(ctx.GasMeter().GasConsumedToLimit(), "gas pool check")
 	}
 
+	// Set newCtx.GasMeter to 0, with a limit of GasWanted (gasLimit)
+	newCtx = ctx.WithGasMeter(sdk.NewGasMeter(gasWanted))
```

Comment:
I think the eth contract call's gas consumption is constrained by the EVM call itself? is it to constraint the gas consumed by cosmos-sdk operations?

Reply:
@yihuang I updated the PR description with more info. There was no way to cap transaction inclusion in the block, because the transaction gas limit was always 0. So, the constraint is for the cosmos-sdk operations.

Reply:
I see, but the gas limit set for eth tx is not for the cosmos-sdk operations.
If we only need to solve tx inclusion issue, maybe it's safer to do this only for checkTx mode?

Reply:
> I see, but the gas limit set for eth tx is not for the cosmos-sdk operations. If we only need to solve tx inclusion issue, maybe it's safer to do this only for checkTx mode?

or only at the end of ante handler in the last handler?

Reply:
I think the evm module will consume gas for all operations like saving variable in state or etc. not only EVM operations.  

May have a risk of mismatch?

Reply:
I think we can set this gas meter in ante handler, but override with a infinite one in msg handler?

Reply:
@yihuang - "maybe it's safer to do this only for checkTx mode" - so do the gas meter replacement only if `ctx.IsCheckTx()` is true.

Reply:
> @yihuang - "maybe it's safer to do this only for checkTx mode" - so do the gas meter replacement only if `ctx.IsCheckTx()` is true.

I was wrong about that, we need to report correct gasWanted in deliver mode too, but we only need to do that in ante handlers, so it's safe to override with an infinite one in msg handler: `Keeper.EthereumTx`, so I suggest to move the `NewInfiniteGasMeter` from `EthSetupContextDecorator` to `Keeper.EthereumTx`.